spec/src/main/asciidoc/rationale.asciidoc

```diff
@@ -0,0 +1,80 @@
+
+* Better failure management as the temporal decoupling would avoid message-loss and avoid cascading failures.
+* Improved elasticity as under fluctuating load the system can decide to scale up and down some of the microservices.
+* The ability to introduce new feature quickly as every component receives and publishes messages.
+
+The MicroProfile Reactive Messaging specification aims to deliver applications embracing the characteristics of reactive systems.
+
+=== On JMS and Message Driven Beans
+
+Java EE offers JMS and Message Driven Beans for handling asynchronous communication; however, there are some problems with these specifications:
+
+* Both are designed for a technology landscape where messaging was a heavyweight process used on the edge of the system to hand control of a transaction from system to another; consequently, there is a lot of boilerplate associated with using these technologies.
+* It is assumed their design that consistency is handled using distributed transactions. 
+However, modern message brokers, popular in microservice deployments, such as Apache Kafka, Amazon Kinesis and Azure Event Hubs, do not support XA transactions, rather, message acknowledgment is handled using offsets with at least once delivery guarantees.
+* They do not have support for asynchronous IO; it is assumed that message processing is done on a single thread, however, many modern specifications are moving to asynchronous IO.
+
+Hence a lighter weight, reactive solution to messaging is desirable for MicroProfile to ensure microservices written using MicroProfile are able to meet the demands required by the architecture.
+
+=== Use cases
+
+MicroProfile Reactive Messaging aims to provide a way to implement _event-driven_ microservices using MicroProfile. 
+The key characteristics of the specification make it versatile and perfectly suited to build different types of architecture and applications.
+
+First, asynchronous interactions with different services and resources can be implemented using Reactive Messaging. 
+Typically, asynchronous database drivers can be used in conjunction with Reactive Messaging to read and write into a data store in a non-blocking and asynchronous manner.
+
+When building microservices, the CQRS and event-sourcing patterns provide an answer to the data sharing between microservices. 
+Reactive Messaging can also be used as the foundation to CQRS and Event-Sourcing mechanism, as these patterns embrace message-passing as core communication pattern.
+
+IOT applications, dealing with events from various devices, and data streaming applications can also be implemented using Reactive Messaging. 
+The application receives events or messages, process them, transform them, and may forward them to another microservices.
+It allows for more fluid architecture for building data-centric applications.
+
+// TODO Pictures for each use case would be nice.
```

Comment:
finished review (on vacation today and until Tuesday but can happily review further commits at odd hours)

Reply:
(Actually I was first back in the office on 29th April.)